File: app/src/main/java.backup/com/lumiyaviewer/lumiya/render/spatial/DrawListEntry.java
package com.lumiyaviewer.lumiya.render.spatial;

import com.lumiyaviewer.lumiya.utils.InlineList;
import com.lumiyaviewer.lumiya.utils.InlineListEntry;
import javax.annotation.Nonnull;

public abstract class DrawListEntry implements InlineListEntry<DrawListEntry> {
    @Nonnull
    final float[] boundingBox = new float[6];
    private volatile InlineList<DrawListEntry> list;
    private DrawListEntry next;
    private DrawListEntry prev;

    public abstract void addToDrawList(@Nonnull DrawList drawList);

    public InlineList<DrawListEntry> getList() {
        return this.list;
    }

    public DrawListEntry getNext() {
        return this.next;
    }

    public DrawListEntry getPrev() {
        return this.prev;
    }

    public void requestEntryRemoval() {
        InlineList inlineList = this.list;
        if (inlineList != null) {
            inlineList.requestEntryRemoval(this);
        }
    }

    public void setList(InlineList<DrawListEntry> inlineList) {
        this.list = inlineList;
    }

    public void setNext(DrawListEntry drawListEntry) {
        this.next = drawListEntry;
    }

    public void setPrev(DrawListEntry drawListEntry) {
        this.prev = drawListEntry;
    }
}

File: Linkpoint/src/main/kotlin/com/linkpoint/render/spatial/DrawListEntry.kt
package com.linkpoint.render.spatial
import java.util.*

import com.linkpoint.utils.InlineList
import com.linkpoint.utils.InlineListEntry
import javax.annotation.Nonnull

abstract class DrawListEntry : InlineListEntry<DrawListEntry> {
    final FloatArray boundingBox = Float[6]
    private volatile InlineList<DrawListEntry> list
    private DrawListEntry next
    private DrawListEntry prev

    public abstract Unit addToDrawList(DrawList drawList)

    public InlineList<DrawListEntry> getList() {
        return this.list
    }

     public fun getNext(): DrawListEntry {
        return this.next
    }

     public fun getPrev(): DrawListEntry {
        return this.prev
    }

    fun requestEntryRemoval() {
        val inlineList: InlineList = this.list
        if (inlineList != null) {
            inlineList.requestEntryRemoval(this)
        }
    }

    fun setList(inlineList: InlineList<DrawListEntry>) {
        this.list = inlineList
    }

    fun setNext(drawListEntry: DrawListEntry) {
        this.next = drawListEntry
    }

    fun setPrev(drawListEntry: DrawListEntry) {
        this.prev = drawListEntry
    }
}
